plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -418,7 +388,25 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
 	return quotas, nil
 }
 
-func (e *quotaEvaluator) evaluate(a admission.Attributes) error {
+func (e *quotaEvaluator) Evaluate(a admission.Attributes) error {
+	e.init.Do(func() {
```

Comment:
apologies david, i ran out of time today to review this, will put it on top
of my agenda tomorrow.

On Wed, Jun 1, 2016 at 8:41 AM, David Eads <notifications@github.com> wrote:

> In plugin/pkg/admission/resourcequota/controller.go
> <https://github.com/kubernetes/kubernetes/pull/26355#discussion_r65352110>
> :
>
> > @@ -418,7 +388,25 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
> >  	return quotas, nil
> >  }
> >
> > -func (e *quotaEvaluator) evaluate(a admission.Attributes) error {
> > +func (e *quotaEvaluator) Evaluate(a admission.Attributes) error {
> > +	e.init.Do(func() {
>
> just curious, but has anyone else adopted this pattern?
>
> Didn't think to look. This just let's me keep a clean API and callers
> don't have to know they need to start the thing once its passed to them. I
> think this one is a little different since it doesn't have a cache to prime.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26355/files/b232d25a8620d4122a0ed2e95026dff6d72e874c#r65352110>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF8dbCumSULiWkTDJNXKQ3EH_96LeVeKks5qHX33gaJpZM4Inkrq>
> .
>